dot/core/service_test.go

```diff
@@ -0,0 +1,651 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package core
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/blocktree"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	mocksruntime "github.com/ChainSafe/gossamer/lib/runtime/mocks"
+	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/lib/trie"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var errTestDummyError = errors.New("test dummy error")
+var testWasmPaths []string
+
+func TestGenerateWasm(t *testing.T) {
+	wasmFilePaths, err := runtime.GenerateRuntimeWasmFile()
+	require.NoError(t, err)
+	testWasmPaths = wasmFilePaths
+}
+
+func Test_Service_StorageRoot(t *testing.T) {
```

Comment:
why is it when I run this test on it's own it fails?
```
go test -timeout 30s -run ^Test_Service_StorageRoot$ github.com/ChainSafe/gossamer/dot/core -v

=== RUN   Test_Service_StorageRoot
=== RUN   Test_Service_StorageRoot/nil_storage_state
=== PAUSE Test_Service_StorageRoot/nil_storage_state
=== RUN   Test_Service_StorageRoot/storage_trie_state_error
=== PAUSE Test_Service_StorageRoot/storage_trie_state_error
=== RUN   Test_Service_StorageRoot/storage_trie_state_ok
=== PAUSE Test_Service_StorageRoot/storage_trie_state_ok
=== CONT  Test_Service_StorageRoot/nil_storage_state
=== CONT  Test_Service_StorageRoot/storage_trie_state_ok
=== CONT  Test_Service_StorageRoot/storage_trie_state_error
=== CONT  Test_Service_StorageRoot/nil_storage_state
    /Users/timothywu/dev/gossamer/dot/core/service.go:148: Unexpected call to *core.MockStorageState.TrieState([<nil>]) at /Users/timothywu/dev/gossamer/dot/core/service.go:148 because:
        expected call at /Users/timothywu/dev/gossamer/dot/core/service_test.go:80 has already been called the max number of times
=== CONT  Test_Service_StorageRoot/storage_trie_state_ok
    /Users/timothywu/dev/gossamer/dot/core/controller.go:137: missing call(s) to *core.MockStorageState.TrieState(is nil) /Users/timothywu/dev/gossamer/dot/core/service_test.go:80
    /Users/timothywu/dev/gossamer/dot/core/controller.go:137: aborting test due to missing call(s)
--- FAIL: Test_Service_StorageRoot (0.00s)
    --- FAIL: Test_Service_StorageRoot/nil_storage_state (0.00s)
    --- PASS: Test_Service_StorageRoot/storage_trie_state_error (0.00s)
    --- FAIL: Test_Service_StorageRoot/storage_trie_state_ok (0.00s)
FAIL
FAIL	github.com/ChainSafe/gossamer/dot/core	0.275s
```

Reply:
hmm odd, this works for me as well

Reply:
Is it a flaky test still now?